### PR TITLE
Refactor Datastore smart contract with libraries

### LINKED ISSUE #40: Refactor Datastore smart contract with libraries
### Description

Our main Datastore contract is now too big to run (out of gas error on deploy). We need to refactor it using libraries. This will allow to deploy the libraries independently from the main contract and will cut gas fees for future users. I'd create a PermissionLibrary, GroupLibrary, FilesLibrary and SettingsLibrary for now. This will also create a more organized backend instead of having just one big file containing all the operations we need.

### PR DESCRIPTION
**Pull request by juslar**
Fix #40

### CODE REVIEW

**Comment by juslar** [comment]
There's no conflicts regarding the `File` struct. I return the same data as we used to but except of returning some of the file's properties directly, I use the new libraries so we don't need to change `createFileFromTuple`.

File: contracts/libraries/GroupLibrary.sol [hunk 1]
```diff
@@ -0,0 +1,129 @@
+pragma solidity ^0.4.18;
+
+import '@aragon/os/contracts/apps/AragonApp.sol';
+import '@aragon/os/contracts/lib/zeppelin/math/SafeMath.sol';
+
+library GroupLibrary {
+    using SafeMath for uint256;
```
[the remark this answers is not in the document]
Comment (juslar) [reply contracts/libraries/GroupLibrary.sol]: No!

File: contracts/libraries/GroupLibrary.sol [hunk 2]
```diff
@@ -0,0 +1,129 @@
...
+    }
+
+    /**
+     * @notice Remove an entity from a group
+     * @param _self GroupData
+     * @param _groupId Id of the group to remove the entity from 
+     * @param _entity Address of the entity
+     */
+    function removeEntityFromGroup(GroupData storage _self, uint _groupId, address _entity) internal {
+        uint indexOfEntity = _self.groups[_groupId].entitiesWithIndex[_entity] - 1;
```
[the remark this answers is not in the document]
Comment (juslar) [reply contracts/libraries/GroupLibrary.sol]: Nice catch :)

File: truffle.js [hunk 3]
```diff
@@ -3,7 +3,7 @@ module.exports = {
     "development": {
       "host": "localhost",
       "port": 8545,
-      "gas": 300000000,
+      //"gas": 300000000,
```
[the remark this answers is not in the document]
Comment (juslar) [reply truffle.js]: I added it to be able to run the project when we had a problem with gas consumption before refactoring our contracts with libraries. This line can be removed now.

**Comment by macor161** [comment]
Perfect! Really nice work! That was a huge refactor!

**Pull request merged**